src/app/heroes/heroes.component.ts

```diff
@@ -36,7 +39,8 @@ export class HeroesComponent implements OnInit {
 
   initHeroes() {
     this.getHeroes()
-      .concat(this.search())
+      .merge(this.search())
+      .merge(this.page())
```

Comment:
Consider getting rid of 3 streams and merge, try using only two streams with concat